src/common/types/vector.cpp

```diff
@@ -550,6 +550,12 @@ Value Vector::GetValueInternal(const Vector &v_p, idx_t index_p) {
 		Value value = child_entries[1]->GetValue(index);
 		return Value::MAP(move(key), move(value));
 	}
+	case LogicalTypeId::UNION: {
+		auto tag = UnionVector::GetTags(*vector)[index]; // ((union_tag_t *)data)[index];
```

Comment:
May need to be careful here - what if the tag vector is a constant? Maybe add a `UnionVector::GetTag(*vector, index)` method or so?